extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -481,6 +491,32 @@ export class BookTocManager implements IBookTocManager {
 		await this._sourceBook.reinitializeContents();
 	}
 
+	/**
+	 * Checks that the targetTreeItem is descendant of the tree item used by the onDrop method.
+	 * @param treeItem The moving element when using dnd.
+	 * @param targetTreeItem The target element where the moving element is dropped.
+	*/
+	isDescendant(treeItem: BookTreeItem, targetTreeItem: BookTreeItem): boolean {
+		return this._usingDragAndDrop && treeItem.rootContentPath === targetTreeItem.rootContentPath && targetTreeItem.book.hierarchyId?.includes(treeItem.book.hierarchyId);
+	}
+
+	/**
+	 * Checks that the book tree item is the parent of the passed element used by the onDrop and the moveTo method.
+	 * @param treeItem The child of the parent tree item.
+	 * @param parentTreeItem The parent of the passed element or the Saved Book tree item.
+	 * @param section (Optional) In case the parentTreeItem is the saved book, verify that the passed Jupyter Book Section is the parent of the treeItem.
+	*/
+	isParent(treeItem: BookTreeItem, parentTreeItem: BookTreeItem, section?: JupyterBookSection): boolean {
+		if (section) {
+			return section.title === treeItem.book.parent?.title && section.file === treeItem.book.parent?.uri;
+		}
+		return JSON.stringify(treeItem.book.parent) === JSON.stringify(parentTreeItem);
```

Comment:
Is there not a better way to compare these than stringifying them? I would expect them to be the same object typically - does the dnd create new clones of the tree items or something?

Reply:
it's not the same object. I'll compare the properties, instead